model_templates/gpu_nemo_textgen/custom_model/custom.py

```diff
@@ -1,25 +1,106 @@
+import logging
 import os
+import subprocess
+from pathlib import Path
+from subprocess import CalledProcessError
+
 import boto3
 from mlpiper.extra.aws_helper import AwsHelper
-from datarobot_drum import RuntimeParameters
 
+from datarobot_drum import RuntimeParameters
+from datarobot_drum.custom_task_interfaces.user_secrets import SecretType
+from datarobot_drum.drum.enum import LOGGER_NAME_PREFIX
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.gpu_predictors import get_optional_parameter
 
 # this path is required by NeMo Inference Server
 NEMO_MODEL_STORE_DIR = "/model-store"
 
 
 def load_model(code_dir: str):
-    print("Downloading model from S3...")
+    load_model_response = (
+        "succeeded"  # a non-empty response is required to signal that load_model succeeded
+    )
+
+    s3_url = get_optional_parameter("s3Url")
+    if s3_url:
+        s3_credential = RuntimeParameters.get("s3Credential")
+        s3_client = S3Client(s3_url, s3_credential)
+
+        keys, dirs = s3_client.list_objects()
+        s3_client.download_files(NEMO_MODEL_STORE_DIR, keys, dirs)
+        return load_model_response
+
+    ngc_registry_url = get_optional_parameter("ngcRegistryUrl")
+    if ngc_registry_url:
+        ngc_credential = RuntimeParameters.get("ngcCredential")
+        ngc_client = NGCRegistryClient(ngc_credential)
+        ngc_client.download_model_version(ngc_registry_url)
+        return load_model_response
+
+    raise Exception(
```

Comment:
I think this should be a `DrumCommonException` right? Otherewise I don't think they get logged appropriately in the actual platform.

Reply:
changed code to raise the DrumCommonException
https://github.com/datarobot/datarobot-user-models/blob/b978bde13f8362d197835f0c42acaaa05e8f055d/custom_model_runner/datarobot_drum/drum/gpu_predictors/utils.py#L74-L85